woocommerce/google-listings-and-ads
# Remove SVGs from JS bundle,

### Changes proposed in this Pull Request:

- Remove SVGs from JS bundle, Use them as `<img>`s.
- Add alt texts.
- Remove unnecessary width, height 100% styles.

Fixes https://github.com/woocommerce/google-listings-and-ads/issues/498

### Screenshots:

See below


### Detailed test instructions:

0. Before pulling this branch
1. Open the following cases in separate tabs to have something to compare `trunk` and `fix/498-svg-out`
2. Check it looks as before (in various window sizes)
3. Check the `alt` text is fine.

#### Cases:

1. **Marketing > GLA (Dashboard) > "Edit" Free Listings > step 2**
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Ffree-listings%2Fedit&pageStep=2`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Ffree-listings%2Fedit&pageStep=2)

	![Hero image](https://user-images.githubusercontent.com/17435/126526974-1e4ca208-eb91-4cd6-9fe0-c1e4d408567b.png)
	> Google Shopping search results example
2. **Marketing > GLA (Dashboard) > "Add paid campaign" > complete setup**
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&guide=campaign-creation-success`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&guide=campaign-creation-success)
	![Successful guide](https://user-images.githubusercontent.com/17435/126527223-0cac74e8-c184-43f5-a1df-7965da57c301.png)
	> Drawing of a person who successfuly launched a campaign

3. Disconnect Google Account using Connection test page 
	[https://gla1.test/wp-admin/admin.php?page=connection-test-admin-page](https://gla1.test/wp-admin/admin.php?page=connection-test-admin-page)

4. **Marketing > GLA**
	[`/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings)
	![Re-connect Google](https://user-images.githubusercontent.com/17435/126528052-f00fb71e-6785-4868-95d0-f4c4ab001c5f.png)
	> Google Logo
5. Disconnect all accounts at [`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings)
6. **Marketing > GLA**
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart)
![Screenshot from 2021-07-21 18-13-02](https://user-images.githubusercontent.com/17435/126528174-aec978ad-b021-45c6-aa55-4dee2845ab6b.png)
	> Google Shopping search results example
	> Drawing of jigsaw puzzles connecting together
	> Drawing of a person looking at their mobile
	> Drawing of a bar and line charts heading up
7. Complete the setup
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed&guide=submission-success`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed&guide=submission-success)
	![Complete setup guide](https://user-images.githubusercontent.com/17435/126528342-8424cba4-b35d-40ba-9e9c-ad675515c20b.png)
	> WooCommerce Logo
	> Google Logo




### Changelog entry

> Tweak - Removed SVGs from JS bundle



### Additional notes:
1. I'm not sure if exposing `/js/src/...` as the assets directory is the safest and fits WordPress conventions. At least it does not require moving files around and lets us keep images close to components that use them.
2. I think that the ideal and the cheapest for an end-user solution would be to inline SVGs into HTML. However, AFAIK we do not use any SSR, and React component instances are 100% JS (as opposed to Custom Elements, which could be served as pure HTML for the first-page load).
3. I made up alt text by myself, so I'd appreciate an approval for the copy (in quote blocks below respective screenshots), @j111q?
4. [According to bundlewatch](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7Cwoocommerce%7CrepoName%7Cgoogle-listings-and-ads%7CrepoCurrentBranch%7Cfix/498-svg-out%7CrepoBranchBase%7Ctrunk%7CcommitSha%7Cc0ecd30c113c42a2e13735f9d4cf0e50f8dc6f4b%7Cresults%7Cstatus%7Cpass%7CfullResults%7CfilePath%7C./js/build/custom-inputs.js%7Csize%7CbaseBranchSize%7CmaxSize%7C./js/build/index.js%7C./js/build/product-attributes.js%7C./js/build/task-complete-setup.js%7C./js/build/index.css%7C./js/build/product-attributes.css%7C./google-listings-and-ads.zip%7Csummary%7CEverything%2Bis%2Bin%2Bcheck%2B(%252B0B,%2B-745.75KB,%2B-7.0%2525)%5EKT%7CKT%7C2DC%7C7MT6%7CFSPT%7C1VF9C%7CD9%7CD9%7C2DC%7CWT%7CWT%7C2DC%7C53X%7C549%7C7WG%7C7Z%7C7Z%7CSG%7C5V9EI%7C63GQ1%7C7HP1C%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7CS%7CI%7CT%7CJ%7CU%5D%7C$F%7CK%7CC%7CD%7CH%7CV%7CI%7CW%7CJ%7CX%5D%7C$F%7CL%7CC%7CD%7CH%7CY%7CI%7CZ%7CJ%7C10%5D%7C$F%7CM%7CC%7CD%7CH%7C11%7CI%7C12%7CJ%7C13%5D%7C$F%7CN%7CC%7CD%7CH%7C14%7CI%7C15%7CJ%7C16%5D%7C$F%7CO%7CC%7CD%7CH%7C17%7CI%7C18%7CJ%7C19%5D%7C$F%7CP%7CC%7CD%7CH%7C1A%7CI%7C1B%7CJ%7C1C%5D%5D%7CQ%7CR%5D%5D) our `index.js` bundle got down by ~52% (719,79KB -> 347.81KB)
